internal/pkg/term/progress/render.go

```diff
@@ -23,52 +22,28 @@ type Renderer interface {
 }
 
 // Render renders r periodically to out until the ctx is canceled or an error occurs.
-// While Render is executing, the terminal cursor is hidden and moved up after each call so that
-// the updates are written in-place.
-func Render(ctx context.Context, out io.Writer, r Renderer) error {
+// While Render is executing, the terminal cursor is hidden and updates are written in-place.
+func Render(ctx context.Context, out WriteFlusher, r Renderer) error {
 	cursor := cursor.NewWithWriter(out)
 	cursor.Hide()
 	defer cursor.Show()
+
 	for {
 		select {
 		case <-ctx.Done():
-			_, err := r.Render(out)
-			return err
+			if _, err := r.Render(out); err != nil {
+				return err
+			}
+			return out.Flush()
 		case <-time.After(renderInterval):
 			nl, err := r.Render(out)
 			if err != nil {
 				return err
 			}
-			cursor.Up(nl + 1) // move the cursor back up to the starting line so that the Renderer is rendered in-place.
+			if err := out.Flush(); err != nil {
+				return err
+			}
+			cursor.Up(nl) // move the cursor back up to the starting line so that the Renderer is rendered in-place.
```

Comment:
Why do we use `nl+1` before and `n` now?

Reply:
It was an off-by-one mistake from me :D 

For example, if we write one line:
```
- The environment stack "phonetool-test" contains your shared resources between services
[] # The cursor is now here
```
So we need to move up by exactly one line, erase it, and rewrite it at that level. Does that make sense?

Reply:
yeah the current behavior makes sense to me but why do we need to move `n+1` before?

Reply:
We didn't it was a bug. I had written this package as a library but it wasn't tested actually using the terminal until this PR. That's when I discovered it :)